src/graphql.js

```diff
@@ -0,0 +1,142 @@
+    disableCard : $ReadOnlyArray<?string>
+|};
+
+const GRAPHQL_URI = '/graphql';
+
+function buildFundingEligibilityVariables() : FundingEligibilityParams {
+    const clientID = getClientID();
+    const merchantID = getMerchantID();
+    const buyerCountry = getBuyerCountry();
+    const currency = getCurrency();
+    const commit = getCommit();
+    const vault = getVault();
+    const intent = getIntent();
+    const disableFunding = getDisableFunding();
+    const disableCard = getDisableCard();
+
+    return {
+        clientID,
+        merchantID,
+        buyerCountry,
+        currency,
+        commit,
+        vault,
+        intent:         intent ? intent.toUpperCase() : intent,
+        disableFunding: disableFunding ? disableFunding.map(f => f && f.toUpperCase()) : disableFunding,
+        disableCard:    disableCard ? disableCard.map(f => f && f.toUpperCase()) : disableCard
+    };
+}
+
+export function callGraphQL<T, V>({ query, variables = {}, headers = {} } : { query : string, variables : V, headers? : { [string] : string } }) : ZalgoPromise<T> {
+    return request({
+        url:     GRAPHQL_URI,
+        method:  'POST',
+        json:    {
+            query,
+            variables
+        },
+        headers: {
+            'x-app-name': 'hosted_fields',
+            ...headers
+        }
+    }).then(({ status, body }) => {
+        const errors = body.errors || [];
+
+        if (errors.length) {
+            const message = errors[0].message || JSON.stringify(errors[0]);
+            throw new Error(message);
+        }
+
+        if (status !== 200) {
+            throw new Error(`${ GRAPHQL_URI } returned status ${ status }`);
+        }
+
+        return body.data;
+    });
+}
+
+// call gql with multiple merchant ids to get fundingEligibility for card
+export function getCardEligibility() : ZalgoPromise<CardEligibility> {
+    const variables = buildFundingEligibilityVariables();
+
+    try {
+        return callGraphQL({
+            query: `
+                query GetFundingEligibility(
+                    $clientID:String,
+                    $merchantID:[ String ],
+                    $buyerCountry:CountryCodes,
+                    $currency:SupportedCountryCurrencies,
+                    $intent:FundingEligibilityIntent,
+                    $commit:Boolean,
+                    $vault:Boolean,
+                    $disableFunding:[ SupportedPaymentMethodsType ],
+                    $disableCard:[ SupportedCardsType ]
+                ) {
+                fundingEligibility(
+                    clientId:$clientID,
+                    buyerCountry:$buyerCountry,
+                    currency:$currency,
+                    intent:$intent,
+                    commit:$commit,
+                    vault:$vault,
+                    disableFunding:$disableFunding,
+                    disableCard:$disableCard,
+                    merchantId:$merchantID
+                ) {
+                    card {
```

Comment:
Can we make

```
                    card {
                        eligible 
                        branded 
                    }
```

a parameter that is passed in by card-components, so anyone can pass in their own query?